src/Sarif.Multitool/Rules/SARIF2001.AuthorHighQualityMessages.cs

```diff
@@ -3,85 +3,106 @@
         {
-            if (messageStrings != null)
+            if (rule.MessageStrings != null && rule.MessageStrings.Count > 0)
             {
                 string messageStringsPointer = reportingDescriptorPointer.AtProperty(SarifPropertyName.MessageStrings);
-
-                foreach (string key in messageStrings.Keys)
+                foreach (KeyValuePair<string, MultiformatMessageString> message in rule.MessageStrings)
                 {
-                    MultiformatMessageString messageString = messageStrings[key];
-
-                    string messageStringPointer = messageStringsPointer.AtProperty(key);
-
-                    AnalyzeMessageString(messageString.Text, messageStringPointer, SarifPropertyName.Text);
+                    AnalyzeMessageString(message.Value.Text, messageStringsPointer.AtProperty(message.Key));
                 }
             }
         }
 
-        protected override void Analyze(MultiformatMessageString multiformatMessageString, string multiformatMessageStringPointer)
+        private void AnalyzeMessageString(string messageString, string messagePointer)
         {
-            AnalyzeMessageString(multiformatMessageString.Text, multiformatMessageStringPointer, SarifPropertyName.Text);
-        }
+            if (string.IsNullOrEmpty(messageString))
+            {
+                return;
+            }
 
-        protected override void Analyze(Message message, string messagePointer)
-        {
-            AnalyzeMessageString(message.Text, messagePointer, SarifPropertyName.Text);
-        }
+            string textPointer = messagePointer.AtProperty(SarifPropertyName.Text);
 
-        private void AnalyzeMessageString(
-            string messageString,
-            string messagePointer,
-            string propertyName)
-        {
-            if (!string.IsNullOrEmpty(messageString) && DoesNotEndWithPeriod(messageString))
+            // IncludeDynamicContent: check if messageString has dynamic content
+            if (!s_dynamicContentRegex.IsMatch(messageString))
+            {
+                // {0}: Placeholder '{1}'
+                LogResult(
+                    textPointer,
+                    nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
+                    messageString
+                    );
+            }
+
+            // EnquoteDynamicContent: check if messageString has enquoted dynamic content
+            if (!s_enquoteDynamicContentRegex.IsMatch(messageString))
             {
-                string textPointer = messagePointer.AtProperty(propertyName);
+                // {0}: Placeholder '{1}'
+                LogResult(
+                    textPointer,
+                    nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_EnquoteDynamicContent_Text),
+                    messageString
+                    );
```

Comment:
>); [](start = 20, length = 2)

Put closing paren on same line as the last argument. You did it correctly on Line 104 below.